variables.tf

```diff
@@ -161,12 +161,12 @@ variable "cloudfront_allowed_cached_methods" {
 }
 
 variable "cloudfront_function_association" {
-  description = "(Optional) Map containing information to associate a function to cloudfront. The first field is `event_type` of the function associated with default cache behavior, it can be viewer-request, viewer-response, origin-request, origin-response. The second field is `function_arn`, the ARN of the function associated with default cache behavior"
-  type = object({
+  description = "(Optional - up to 2 per distribution) List containing information to associate a CF function to cloudfront. The first field is `event_type` of the CF function associated with default cache behavior, it can be viewer-request or viewer-response"
```

Comment:
Is there any reason to say "up to 2" per distribution? Given the way the variable is declared you can add as many as you want. If that is an actual restriction from AWS maybe there should be some control to avoid adding more than two in the website.tf file.

Reply:
You can only have 2 CF functions per distribution, one for request and one for response. Not sure we can validate a max of 2 within tf that's why I put it in description. I guess a tf plan will do the validation and fail